editor/src/components/canvas/controls/position-outline.tsx

```diff
@@ -0,0 +1,148 @@
+    if (element != null && isRight(element.element) && isJSXElement(element.element.value)) {
+      return right(element.element.value.props)
+    } else {
+      return left(element?.props ?? {})
+    }
+  }, 'usePropsOrJSXAttributes')
+}
+
+const useContainingFrameForElement = (path: ElementPath): CanvasRectangle | null => {
+  return useEditorState((store) => {
+    const containingBlockPath = MetadataUtils.findContainingBlock(store.editor.jsxMetadata, path)
+    if (containingBlockPath != null) {
+      return MetadataUtils.getFrameInCanvasCoords(containingBlockPath, store.editor.jsxMetadata)
+    } else {
+      return null
+    }
+  }, 'useContainingFrameForElement')
+}
+
+const collectPinOutlines = (
+  attributes: PropsOrJSXAttributes,
+  frame: CanvasRectangle,
+  containingFrame: CanvasRectangle,
+  canvasOffset: CanvasPoint,
+): PinOutlineProps[] => {
+  const pinLeft = eitherToMaybe(getLayoutProperty('PinnedLeft', attributes))
+  const pinTop = eitherToMaybe(getLayoutProperty('PinnedTop', attributes))
+  const pinRight = eitherToMaybe(getLayoutProperty('PinnedRight', attributes))
+  const pinBottom = eitherToMaybe(getLayoutProperty('PinnedBottom', attributes))
+  let pins: PinOutlineProps[] = []
+  if (pinLeft != null) {
+    pins.push({
+      key: 'left',
+      isHorizontalLine: true,
+      size: frame.x - containingFrame.x,
+      startX: containingFrame.x + canvasOffset.x,
+      startY: frame.y + frame.height / 2 + canvasOffset.y,
+    })
+  }
+  if (pinTop != null) {
+    pins.push({
+      key: 'top',
+      isHorizontalLine: false,
+      size: frame.y - containingFrame.y,
+      startX: frame.x + frame.width / 2 + canvasOffset.x,
+      startY: containingFrame.y + canvasOffset.y,
+    })
+  }
+  if (pinRight != null) {
+    pins.push({
+      key: 'right',
+      isHorizontalLine: true,
+      size: containingFrame.x + containingFrame.width - (frame.x + frame.width),
+      startX: frame.width + frame.x + canvasOffset.x,
+      startY: frame.y + frame.height / 2 + canvasOffset.y,
+    })
+  }
+  if (pinBottom != null) {
+    pins.push({
+      key: 'bottom',
+      isHorizontalLine: false,
+      size: containingFrame.y + containingFrame.height - (frame.y + frame.height),
+      startX: frame.x + frame.width / 2 + canvasOffset.x,
+      startY: frame.height + frame.y + canvasOffset.y,
+    })
+  }
+  return pins
+}
+
+interface PinOutlineProps {
+  key: string
+  isHorizontalLine: boolean
+  startX: number
+  startY: number
+  size: number
+}
+
+const PinOutline = betterReactMemo(
+  'PinOutline',
+  (props: PinOutlineProps): JSX.Element => {
+    const colorTheme = useColorTheme()
+    const width = props.isHorizontalLine ? props.size : 0
+    const height = props.isHorizontalLine ? 0 : props.size
+    const borderTop = props.isHorizontalLine ? `1px dashed ${colorTheme.primary.value}` : 'none'
+    const borderLeft = props.isHorizontalLine ? 'none' : `1px dashed ${colorTheme.primary.value}`
+    return (
+      <>
```

Comment:
Looks like this fragment is no longer necessary

Reply:
fixed